core/workspace_comment_svg.js

```diff
@@ -0,0 +1,591 @@
+        /** @type {!Element} */ (this.svgGroup_), 'blocklyDraggable');
+  } else {
+    Blockly.utils.removeClass(
+        /** @type {!Element} */ (this.svgGroup_), 'blocklyDraggable');
+  }
+};
+
+/**
+ * Set whether this comment is movable or not.
+ * @param {boolean} movable True if movable.
+ * @package
+ */
+Blockly.WorkspaceCommentSvg.prototype.setMovable = function(movable) {
+  Blockly.WorkspaceCommentSvg.superClass_.setMovable.call(this, movable);
+  this.updateMovable();
+};
+
+/**
+ * Recursively adds or removes the dragging class to this node and its children.
+ * @param {boolean} adding True if adding, false if removing.
+ * @package
+ */
+Blockly.WorkspaceCommentSvg.prototype.setDragging = function(adding) {
+  if (adding) {
+    var group = this.getSvgRoot();
+    group.translate_ = '';
+    group.skew_ = '';
+    Blockly.utils.addClass(
+        /** @type {!Element} */ (this.svgGroup_), 'blocklyDragging');
+  } else {
+    Blockly.utils.removeClass(
+        /** @type {!Element} */ (this.svgGroup_), 'blocklyDragging');
+  }
+};
+
+/**
+ * Return the root node of the SVG or null if none exists.
+ * @return {Element} The root SVG node (probably a group).
+ * @package
+ */
+Blockly.WorkspaceCommentSvg.prototype.getSvgRoot = function() {
+  return this.svgGroup_;
+};
+
+/**
+ * Returns this comment's text.
+ * @return {string} Comment text.
+ * @package
+ */
+Blockly.WorkspaceCommentSvg.prototype.getContent = function() {
+  return this.textarea_ ? this.textarea_.value : this.content_;
+};
+
+/**
+ * Set this comment's content.
+ * @param {string} content Comment content.
+ * @package
+ */
+Blockly.WorkspaceCommentSvg.prototype.setContent = function(content) {
+  Blockly.WorkspaceCommentSvg.superClass_.setContent.call(this, content);
+  if (this.textarea_) {
+    this.textarea_.value = content;
+  }
+};
+
+/**
+ * Update the cursor over this comment by adding or removing a class.
+ * @param {boolean} enable True if the delete cursor should be shown, false
+ *     otherwise.
+ * @package
+ */
+Blockly.WorkspaceCommentSvg.prototype.setDeleteStyle = function(enable) {
+  if (enable) {
+    Blockly.utils.addClass(
+        /** @type {!Element} */ (this.svgGroup_), 'blocklyDraggingDelete');
+  } else {
+    Blockly.utils.removeClass(
+        /** @type {!Element} */ (this.svgGroup_), 'blocklyDraggingDelete');
+  }
+};
+
+Blockly.WorkspaceCommentSvg.prototype.setAutoLayout = function() {
+  // NOP for compatibility with the bubble dragger.
+};
+
+/**
+ * Decode an XML comment tag and create a rendered comment on the workspace.
+ * @param {!Element} xmlComment XML comment element.
+ * @param {!Blockly.Workspace} workspace The workspace.
+ * @param {number=} opt_wsWidth The width of the workspace, which is used to
+ *     position comments correctly in RTL.
+ * @return {!Blockly.WorkspaceCommentSvg} The created workspace comment.
+ * @package
+ */
+Blockly.WorkspaceCommentSvg.fromXml = function(xmlComment, workspace,
+    opt_wsWidth) {
```

Comment:
Does it make sense to call the superclass fromXml first and do the rendering specific stuff after?

Reply:
No, because the moveBy can depend on the size of the workspace itself if it's rendered.

The model we have for building blocks (using the headless constructor and then adding stuff if it's rendered) has led to a lot of issues with initialization and ordering, even though it looks like a tidier way to do it.

I'm avoiding that model here, and ideally will eventually fix the block initialization as well.